### PR TITLE
refactor and split input fn

### LINKED ISSUE #68: `cosmic_edit_bevy_events` has too many args
Needs a refactor, I'm going to give it a shot.

@StaffEngineer Do you think putting input in a separate file like `cursor.rs` is a good idea?

### PR DESCRIPTION
**Pull request by bytemunch**
closes #68 

`cosmic_edit_bevy_events` has been decoupled into `input_mouse` and `input_kb`, and moved to `input.rs`

Moved `lib.rs` around, ordered as enums > structs > plugin > functions

would appreciate a macOS test please @StaffEngineer :)

### CODE REVIEW

**Comment by Dimchikkk** [comment]
image

selection is broken... on main I can select text, on this code changes I can't :/

File: src/input.rs [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+        return;
+    }
+
+    let primary_window = windows.single();
+    let scale_factor = primary_window.scale_factor() as f32;
+    let (camera, camera_transform) = camera_q.iter().find(|(c, _)| c.is_active).unwrap();
+    for (mut editor, node_transform, text_position, entity, x_offset, node_opt, sprite_opt) in
+        &mut cosmic_edit_query.iter_mut()
+    {
+        if active_editor.0 != Some(entity) {
```
Comment (Dimchikkk) [inline src/input.rs:47]: User still should be able to select text on non-active editors.

File: src/lib.rs [hunk 2]
```diff
@@ -420,6 +419,7 @@ impl Plugin for CosmicEditPlugin {
                 swash_cache: SwashCache::new(),
             })
             .insert_resource(CosmicFontSystem(font_system))
+            .insert_resource(ClickTimer(Timer::from_seconds(0.5, TimerMode::Once)))
```
Comment (Dimchikkk) [inline src/lib.rs]: Is this for https://github.com/StaffEngineer/bevy_cosmic_edit/issues/45 ? :)

Comment (bytemunch) [reply src/lib.rs]: Oh yeah forgot to swap that to another branch lol

File: src/input.rs (hunk 1 above)
Comment (bytemunch) [reply src/input.rs:47]: Ah yep I'll re-add the readonly bits

**Comment by Dimchikkk** [comment]
Cool, so if editor has readonly text should be selectable if there is no readonly and it's not active - it shouldn't.... this is gonna be important for placeholder implementation.

**Comment by bytemunch** [comment]
Yeah, currently the mouse just ignores readonly as mouse doesn't write to buffer. Placeholder should be possible with this implementation, if as soon as it's active the placeholder text will get deleted. Holding placeholder on empty buffer might need a bit of wrangling but there's always a way

**Review by Dimchikkk — APPROVED** [review]

**Pull request merged**